svm/src/transaction_processor.rs

```diff
@@ -309,62 +296,112 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             program_cache_for_tx_batch
         });
 
-        let (loaded_transactions, load_accounts_us) = measure_us!(load_accounts(
+        let mut account_loader = AccountLoader::new(
             callbacks,
-            sanitized_txs,
-            validation_results,
-            &mut error_metrics,
             config.account_overrides,
-            &environment.feature_set,
-            environment
-                .rent_collector
-                .unwrap_or(&RentCollector::default()),
-            &program_cache_for_tx_batch,
-        ));
+            program_cache_for_tx_batch,
+        );
 
         let enable_transaction_loading_failure_fees = environment
             .feature_set
             .is_active(&enable_transaction_loading_failure_fees::id());
-        let (processing_results, execution_us): (Vec<TransactionProcessingResult>, u64) =
-            measure_us!(loaded_transactions
-                .into_iter()
-                .zip(sanitized_txs.iter())
-                .map(|(load_result, tx)| match load_result {
-                    TransactionLoadResult::NotLoaded(err) => Err(err),
-                    TransactionLoadResult::FeesOnly(fees_only_tx) => {
-                        if enable_transaction_loading_failure_fees {
-                            Ok(ProcessedTransaction::FeesOnly(Box::new(fees_only_tx)))
-                        } else {
-                            Err(fees_only_tx.load_error)
-                        }
+
+        let (mut validate_fees_us, mut load_us, mut execution_us): (u64, u64, u64) = (0, 0, 0);
+
+        // Validate, execute, and collect results from each transaction in order.
+        // With SIMD83, transactions must be executed in order, because transactions
+        // in the same batch may modify the same accounts. Transaction order is
+        // preserved within entries written to the ledger.
+        for (tx, check_result) in sanitized_txs.iter().zip(check_results) {
+            let (validate_result, single_validate_fees_us) = measure_us!(check_result
```

Comment:
nit: can you introduce a new function called `validate_transaction` which wraps `validate_transaction_nonce` and `validate_transaction_fee_payer`? There's a lot of nesting here that makes the code hard to read IMO